### ISSUE TITLE
code_warntype doesn't document the optimize kwarg

### ISSUE THREAD

**Issue by oxinabox** [JuliaLang/julia#48804]
`@code_warntype` takes an optimize argument that seems to default to `false`.
This is never mentioned in it's docstring, not the docsting for the function form.
You can see it clearly exists in the difference between the following two examples.

```julia
julia> @code_warntype (()->1+1)()
MethodInstance for (::var"#7#8")()
  from (::var"#7#8")() @ Main REPL[16]:1
Arguments
  #self#::Core.Const(var"#7#8"())
Body::Int64
1 ─ %1 = (1 + 1)::Core.Const(2)
└──      return %1


julia> @code_warntype optimize=true (()->1+1)()
MethodInstance for (::var"#9#10")()
  from (::var"#9#10")() @ Main REPL[17]:1
Arguments
  #self#::Core.Const(var"#9#10"())
Body::Int64
1 ─     return 2
```

I assume it is the same as the one which is [documented for `@code_typed`](https://docs.julialang.org/en/v1/stdlib/InteractiveUtils/#InteractiveUtils.@code_typed)

**Comment by oscardssmith** [comment]
wait really? I though `@code_warntype` always showed post-optimization.

**Comment by oxinabox** [comment]
It seems not to. In particular, if you pass `optimize=true` inlining happens, and otherwise it does not.
Possibly that is why it doesn't by default, since how to fix type-instability is clearer if you don't inline things. (And more generally it is more readable if you don't inline things)

**Comment by aviatesk** [comment]
> Possibly that is why it doesn't by default, since how to fix type-instability is clearer if you don't inline things. (And more generally it is more readable if you don't inline things)

You're right. The default value of `optimize` should be `false` for `code_warntype`, but its' definitely better to mention it in its docstring.
